src/lib.rs

```diff
@@ -70,30 +127,11 @@ fn pid1_handling(settings: &Pid1Settings, child: Option<Child>) -> ! {
     loop {
         for signal in signals.forever() {
             if signal == SIGTERM || signal == SIGINT {
-                let exit_code = signal + 128;
-                match child {
-                    Some(child_pid) => {
-                        let pid = Pid::from_raw(child_pid as i32);
-                        let nix_signal = if signal == SIGTERM {
-                            nix::sys::signal::SIGTERM
-                        } else {
-                            nix::sys::signal::SIGINT
-                        };
-                        let result = kill(pid, Some(nix_signal));
-                        match result {
-                            Ok(()) => std::process::exit(exit_code),
-                            Err(errno) => {
-                                if settings.log {
-                                    eprintln!(
-                                        "pid1-rs: kill() failed on {pid} with errno: {errno}"
-                                    );
-                                }
-                                std::process::exit(exit_code)
-                            }
-                        }
-                    }
-                    None => std::process::exit(exit_code),
-                }
+                // We also do graceful exit in a separate thread so that
+                // pid1 exits as soon as possible
+                let _ = std::thread::spawn(move || gracefull_exit(settings, signal, child));
```

Comment:
This has the potential to spawn this graceful exit thread twice. I'd mentioned the possibility of a `oneshot`, but a simpler solution may be sufficient: keeping a mutable variable in this function `let mut is_exiting = false§ and then, here, checking is `is_exiting` is false before spawning. When spawning you'd need to set `is_exiting` to true.

Reply:
Yeah, just like how you mentioned previously. :-) Thanks, have addressed it using an enum.